lib/fizzy/execute.hpp

```diff
@@ -20,7 +20,7 @@ struct ExecutionResult
 {
     const bool trapped = false;
     const bool has_value = false;
-    const Value value = 0;
+    const Value value{};
```

Comment:
@chfast After this default-initialization which member of `Value` is it allowed to access without UB?

Reply:
> If T is a union type, the first non-static named data member is zero-initialized and all padding is initialized to zero bits. 

Maybe that is not the best idea in the end.

Reply:
So what should be done here?

Reply:
I did some experiments and this is fine.

Reply:
At some point documentation for `ExecutionResult` shoud be added, explaining how it is allowed to access `value` (only when  `!trapped & has_value` and only as the correct type)